VirtoCommerce.Storefront.Model/Common/MutablePagedList.cs

```diff
@@ -238,14 +238,19 @@ public ICollection Keys
         {
             get
             {
-                throw new NotImplementedException();
+               TryReloadPagedData();
+               var dict = _pagedList.OfType<IAccessibleByIndexKey>().ToDictionary(x => x.IndexKey, x => x);
```

Comment:
.OfType can throw the null exception. So you should be careful with this operator in Generics.
For instance, the Breadcrumb is not implement the IAccessibleByIndexKey :)
I recommend to check the null reference and return something like this:
return Enumerable.Empty<string>().ToDictionary(x => x, x => x);

Reply:
the name 'dict' is not necessary should be short form in this context. Better name 'dictionary', it's more clear

Reply:
in the TryReloadPagedData method is located the hidden initialization of the _pagedList. It is wrong because you should to go and view this method to catch on the context

Reply:
'dict' is renamed to 'dictionary'.